package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -699,15 +711,25 @@ def _filter_atoms(self, donors, acceptors):
     def _prepare(self):
         self.results.hbonds = [[], [], [], [], [], []]
 
+        # Set unpaired acceptor and hydrogen AtomGroups
+        # self.acceptors_ag = self.guess_acceptors() #NOTE: do I need this?
+        # self.hydrogens_ag = self.guess_hydrogens()
+        # self.donors_ag = self.guess_donors()
+
         # Set atom selections if they have not been provided
         if self.acceptors_sel is None:
-            self.acceptors_sel = self.guess_acceptors()
+            self.acceptors_ag = self.guess_acceptors()
+            self.acceptors_sel = self._group_categories(self.acceptors_ag) #NOTE: do I need this?
         if self.hydrogens_sel is None:
-            self.hydrogens_sel = self.guess_hydrogens()
+            self.hydrogens_ag = self.guess_hydrogens()
+            self.hydrogens_sel = self._group_categories(self.hydrogens_ag)
+        #NOTE: add in self.donors_sel? Is this not done for a reason?
+        if self.donors_sel is None:
+            self.donors_ag = self.guess_donors()
+            self.donors_sel = self._group_categories(self.donors_ag)
 
         # Select atom groups
-        self._acceptors = self.u.select_atoms(self.acceptors_sel,
-                                              updating=self.update_selections)
+        self._acceptors = self.guess_acceptors()
```

Comment:
You already have `self.acceptors_ag` above so you don't need to guess it again. Simply parse `self._acceptors = self.acceptors_ag`

Reply:
Oops I missed this one, thank you for picking it up :)